File: 70.Climbing stairs.py
class Solution:
    def climbStairs(self,n:int)->int:
        if n==1: return 1
        if n==2: return 2
        first, second = 1, 2

        for _ in range(3,n+1):
            third = first+second
            first, second = second, third
        return second

File: 70. Climbing stairs.java
class Solution{
    public int climbStairs(int n){
        if(n==1) return 1;
        if(n==2) return 2;
        int first = 1, second = 2;

        for(i=3;i<=n; i++){
            int third = first+second;
            first = second;
            second= third;
        }
        return second;
    }
}